includes/OAuth/OAuthAuthenticator.php

```diff
@@ -82,7 +82,7 @@ public function updateAuthTokenIfNecessary(): void
     {
         $currentToken = $this->settingsRepository->getAccessToken();
 
-        if ($currentToken->isValid()) {
+        if (!is_null($currentToken) && $currentToken->isValid()) {
```

Comment:
when could it be null? unserealize() would return the value or false i believe

Reply:
Yes the unserialize method could return false if the data could not be deserialized. This behaviour is abstracted away in the settings repository. It will return the token if a valid token is available. If no token is available or deserializing the token was not successful it will return null since it does not make a difference which of the two cases happend. We have to request a new token anyway.

Reply:
so we add this to the condition to prevent making a call to the isValid method if the $currentToken is null ?

Reply:
Yes, that is correct.